keps/sig-node/127-user-namespaces/README.md

```diff
@@ -363,88 +549,463 @@ Note this section is a WIP yet.
+    heartbeats, leader election, etc.)
+-->
 
-* **Will enabling / using this feature result in increasing size or count of
-the existing API objects?** Yes. The PodSpec will be increased.
+###### Will enabling / using this feature result in introducing new API types?
 
-* **Will enabling / using this feature result in increasing time taken by any
-operations covered by [existing SLIs/SLOs]?**
+<!--
+Describe them, providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+-->
 
-* **Will enabling / using this feature result in non-negligible increase of
-resource usage (CPU, RAM, disk, IO, ...) in any components?**: No.
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+<!--
+Describe them, providing:
+  - Which API(s):
+  - Estimated increase:
+-->
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes. The pod.Spec.HostUsers field is a bool, should be small.
+
+<!--
+Describe them, providing:
+  - API type(s):
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+-->
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Not in any Kubernetes component, it might take more time for the container
+runtime to start the pod _if they support old kernels_.
+
+The SLO that might be affected is:
+
+> Startup latency of schedulable stateless pods, excluding time to pull images and run init containers, measured from pod creation timestamp to when all its containers are reported as started and observed via watch, measured as 99th percentile over last 5 minutes
+
+The rootfs needs to be accessible by the user in the user namespace the pod is.
+As every pod might run as a different user, we only know the mapping for a pod
+when it is created.
+
+If new kernels are used (5.19+), the container runtime can use idmapped mounts
+in the rootfs and the UID/GID shifting needed so the pod can access the rootfs
+takes a few ms (it is just a bind mount).
+
+If the container runtimes want to support old kernels too, they will need to
+chown the rootfs to the new UIDs/GIDs. This can be slow, although the metacopy
+overlayfs parameter helps so it is not so bad.
+
+The options we have for this plumbing to setup the rootfs:
+ * Consider it the same way we consider the image pull, and explicitelly exclude
+   it from the SLO.
+ * Tell runtimes to not support old kernels (KEP authors are working in the
+   containerd and CRI-O implementation too, we can easily do this)
+ * Tell runtimes to support old kernels if they want, but with a big fat warning
+   in the docs about the implications
+
+In any case, the kubernetes components do not need any change.
```

Comment:
This is great - thank you!

I would be inclined to saying that pods using userns are out-of-scope for the SLO for old kernels, but in general, I really like this section.

Reply:
Thanks! 

I've added this text to the section (at the top), let me know if you prefer to change it or anything else:

```
With the info below, we decided that container runtimes _can_ support old
kernels, but the SLO will be excluded in that case.
```